src/vector_of_array.jl

```diff
@@ -0,0 +1,65 @@
+abstract AbstractVectorOfArray{T, N} <: AbstractArray{T, N}
+
+# Based on code from M. Bauman Stackexchange answer + Gitter discussion
+type VectorOfArray{T, N, A} <: AbstractVectorOfArray{T, N}
+  data::A # A <: AbstractVector{<: AbstractArray{T, N - 1}}
+  dims::NTuple{N, Int}
```

Comment:
The only issue is can I make the field readonly? Might cause strange issues if the user mutates the dim field.